app/controllers/staff/speakers_controller.rb

```diff
@@ -42,19 +32,19 @@ def edit
   def update
     @speaker = Speaker.find(params[:id])
     if @speaker.update(speaker_params)
-      redirect_to event_staff_speaker_url(@event, @speaker)
+      redirect_to event_staff_speakers_path(current_event)
     else
       render :edit
     end
   end
 
   def destroy
     @speaker = Speaker.find_by!(id: params[:id])
-    proposal = speaker.proposal
+    program_session = speaker.program_session
```

Comment:
I believe you can remove this line and reference it via speaker.program_session in the message below.